xdp/src/netlink.rs

```diff
@@ -133,10 +183,116 @@ impl NetlinkSocket {
 
         Ok(messages)
     }
+
+    /// Opens a listener socket for netlink updates
+    /// NETLINK_ROUTE socket subscribed to `groups` bitmask
+    pub fn open_multicast_listener(groups: u32) -> Result<Self, io::Error> {
+        let raw = unsafe { socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE) };
+        if raw < 0 {
+            return Err(io::Error::last_os_error());
+        }
+        let sock = unsafe { OwnedFd::from_raw_fd(raw) };
+
+        // Subscribe to multicast groups
+        let mut addr: sockaddr_nl = unsafe { mem::zeroed() };
+        addr.nl_family = AF_NETLINK as u16;
+        addr.nl_groups = groups;
+        if unsafe {
+            bind(
+                sock.as_raw_fd(),
+                &addr as *const _ as *const _,
+                mem::size_of::<sockaddr_nl>() as u32,
+            )
+        } < 0
+        {
+            return Err(io::Error::last_os_error());
+        }
+
+        // Larger rcvbuf to survive bursts of updates
+        let rcvbuf: i32 = 1 << 20;
```

Comment:
Can you say more about this? Is it really necessary? We have a thread that polls
this socket in a tight loop, I'd be surprised if it was overflowing.